### PR TITLE
seastore: RBM clean-ups

### PR DESCRIPTION
**Pull request by myoungwon**
Signed-off-by: Myoungwon Oh <[myoungwon.oh@samsung.com](mailto:myoungwon.oh@samsung.com)>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/os/seastore/CMakeLists.txt [hunk 1]
```diff
@@ -36,7 +36,7 @@ set(crimson_seastore_srcs
   extent_placement_manager.cc
   object_data_handler.cc
   seastore.cc
-  random_block_manager/nvme_manager.cc
+  random_block_manager/block_nvme_manager.cc
```
Comment (cyx1231st) [inline src/crimson/os/seastore/CMakeLists.txt:39]: block_nvme_manager.cc -> block_rbm_manager.cc

also the commit title should be `rename NVMeManager to BlockRBMManager`

File: src/crimson/os/seastore/random_block_manager/block_nvme_manager.cc [hunk 2]
```diff
@@ -10,16 +10,16 @@
 #include "nvmedevice.h"
 #include "include/interval_set.h"
 #include "include/intarith.h"
-#include "nvme_manager.h"
+#include "block_nvme_manager.h"
```
Comment (cyx1231st) [inline src/crimson/os/seastore/random_block_manager/block_nvme_manager.cc:13]: block_nvme_manager.h -> block_rbm_manager.h

File: src/crimson/os/seastore/random_block_manager/nvme_manager.cc [hunk 3]
```diff
@@ -62,28 +57,28 @@ NVMeManager::mkfs_ertr::future<> NVMeManager::initialize_blk_alloc_area() {
...
     super.start_alloc_area / super.block_size
-  ).safe_then([this, b_block, start] () mutable {
+  ).safe_then([this, b_block, start, FNAME] () mutable {
 
     /* initialize bitmap blocks as unused */
     auto max = max_block_by_bitmap_block();
     auto max_block = super.size / super.block_size;
     blk_no_t end = round_up_to(max_block, max) - 1;
-    logger().debug(" init start {} end {} ", start, end);
+    DEBUG(" init start {} end {} ", start, end);
```
Comment (cyx1231st) [inline src/crimson/os/seastore/random_block_manager/nvme_manager.cc:66]: Drop the first unnecessary space, also fix the other logs.

File: src/crimson/os/seastore/random_block_manager/nvme_manager.cc [hunk 4]
```diff
@@ -62,28 +57,28 @@ NVMeManager::mkfs_ertr::future<> NVMeManager::initialize_blk_alloc_area() {
 
   return rbm_sync_block_bitmap(b_block,
     super.start_alloc_area / super.block_size
-  ).safe_then([this, b_block, start] () mutable {
+  ).safe_then([this, b_block, start, FNAME] () mutable {
```
Comment (cyx1231st) [inline src/crimson/os/seastore/random_block_manager/nvme_manager.cc:60]: our convention is `[...]() mutable {` or `[...] {`.

`mutable` is also not needed here.

File: src/crimson/os/seastore/random_block_manager/rbm_device.h [hunk 5]
```diff
@@ -342,67 +303,37 @@ class NVMeBlockDevice : public Device {
...
 };
 
 /*
  * Implementation of NVMeBlockDevice with POSIX APIs
  *
- * PosixNVMeDevice provides NVMe SSD interfaces through POSIX APIs which is
+ * NVMeBlockDevice provides NVMe SSD interfaces through POSIX APIs which is
  * generally available at most operating environment.
  */
-class PosixNVMeDevice : public NVMeBlockDevice {
```
Comment (cyx1231st) [inline src/crimson/os/seastore/random_block_manager/rbm_device.h:402]: Commit title is wrong:
...
2. PosixNVMeDevice to NVMeBlockDevice

**Comment by cyx1231st** [comment]
refer to https://github.com/ceph/ceph/pull/46986#issuecomment-1182205054

@zhscn

**Comment by zhscn** [comment]
`HDDBlockDevice` will inherit `RBMDevice` after merge this pr.

File: src/crimson/os/seastore/random_block_manager/rbm_device.h (hunk 5 above)
Comment (myoungwon) [reply src/crimson/os/seastore/random_block_manager/rbm_device.h:402]: -class PosixNVMeDevice : public NVMeBlockDevice {
+class NVMeBlockDevice : public RBMDevice {

PosixNVMeDevice was removed, then renamed NVMeBlockDevice.

**Comment by cyx1231st** [comment]
jenkins test api

File: src/crimson/os/seastore/random_block_manager/rbm_device.h (hunk 5 above)
Comment (cyx1231st) [reply src/crimson/os/seastore/random_block_manager/rbm_device.h:402]: I meant the commit message is wrong:

> 2. rename PosixNVMeBlockDevice to NVMeBlockDevice

Should be `2. PosixNVMeDevice to NVMeBlockDevice`

File: src/crimson/os/seastore/random_block_manager/nvme_block_device.cc [hunk 6]
```diff
@@ -20,7 +20,7 @@ namespace {
 
 namespace crimson::os::seastore::nvme_device {
```
Comment (cyx1231st) [inline src/crimson/os/seastore/random_block_manager/nvme_block_device.cc:21]: Namespace nvme_device looks inappropriate.

Probably rename to random_block_device or block_device ?

File: src/crimson/os/seastore/random_block_manager/rbm_device.h [hunk 7]
```diff
@@ -306,146 +169,6 @@ class RBMDevice : public Device {
...
-  uint32_t stream_id_count = 1; // stream is disabled, defaultly.
-  uint32_t awupf = 0;
-
-  uint64_t write_granularity = 4096;
-  uint64_t write_alignment = 4096;
-  uint32_t atomic_write_unit = 4096;
-
-  bool data_protection_enabled = false;
-};
-
```
Comment (cyx1231st) [inline src/crimson/os/seastore/random_block_manager/rbm_device.h]: Some includes are nolonger necessary after moving them out.

**Review by cyx1231st — APPROVED** [review]

**Comment by myoungwon** [comment]
jenkins retest this please

**Comment by cyx1231st** [comment]
jenkins retest this please

**Comment by myoungwon** [comment]
jenkins test make check

**Pull request merged**